plugins/queue/scheduler.js

```diff
@@ -203,13 +207,15 @@ async function startPeriodic(executor, config) {
             // eslint-disable-next-line max-len
             if (err && err.message !== 'Job already enqueued at this time with same arguments') {
                 shouldRetry = true;
+                logger.error(`duplicate build: failed to enqueue for job ${job.id}: ${err}`);
             }
         }
         if (!shouldRetry) {
             return Promise.resolve();
```

Comment:
there should be ` logger.info` here if there is no error. I think it's better to inverse the check and move the retry code inside here.